contracts/docs/international-harbour.md

```diff
@@ -0,0 +1,149 @@
+# International Harbour
+
+**Permissionless, append-only registry for Safe transactions and ECDSA signatures**
+
+International Harbour (implemented by the `SafeInternationalHarbour` contract) provides an on-chain, chain-agnostic store for Safe (multisig) transactions and their signatures. Clients can reconstruct the full multisig payload with only the Safe address, target chain ID, nonce, and owner set—no off-chain indexer required.
+
+## Motivation
+
+Off-chain coordination of Safe transactions (e.g. collecting ECDSA signatures) is brittle without a reliable broadcast and indexing layer. International Harbour solves this by:
+
+- Allowing any EOA to publish a Safe transaction and its signature on-chain.
+- Persisting each unique transaction (by its EIP-712 digest) exactly once and appending signatures under signer-specific keys.
+- Enabling on-chain lookups and client-side reconstruction of the full payload.
+
+## Key Disclaimers
+
+- **Signature malleability**: The contract does _not_ enforce EIP-2 low-`s`. If two `(r,s,v)` tuples yield the same signer, both are stored; client code must dedupe if needed.
+- **Parameter collision**: Transactions are identified _solely_ by `safeTxHash`. In the astronomically unlikely event of a hash collision, the first stored parameters prevail; later submissions are ignored.
+- **EOA-only signers**: Only ECDSA signatures from EOAs are supported. Contract-based signers (e.g. ERC-1271) cannot be verified on-chain in a chain-agnostic way.
+
+## Implementation Details
+
+### Data Structures
+
+- **SafeTransaction**: Mirror of Safe's `SafeTx` struct:
+
+  ```solidity
+  struct SafeTransaction {
+    address to;
+    uint256 value;
+    uint8 operation;
+    uint256 safeTxGas;
+    uint256 baseGas;
+    uint256 gasPrice;
+    address gasToken;
+    address refundReceiver;
```

Comment:
I would have an `SafeTransactionGas` struct indirection here - where you can point avoid writing 5 additional slots if they are all 0.